asserts/pool.go

```diff
@@ -579,25 +775,30 @@ func (p *Pool) unresolvedBookkeeping() {
 	//  * in error
 	//  * unchanged
 	//  * or unresolved
-	for uniq, u := range p.unresolved {
-		e := u.err
-		if e == nil {
-			if u.at.Revision == RevisionNotKnown {
-				e = ErrUnresolved
-			} else {
-				// unchanged
-				p.unchanged[uniq] = true
+	processUnresolved := func(unresolved map[string]unresolvedAssertRecord) {
+		for uniq, ur := range unresolved {
+			e := ur.error()
+			if e == nil {
+				if ur.revision() == RevisionNotKnown {
```

Comment:
is this the only usage of .revision() ? if that's the case maybe we can make it more specific

Reply:
Indeed. Made it more specific (isRevisionNotKnown() bool)